### PR TITLE
[feature] Adjust NPC Dialog Box and add Temporary Dialogue Text

### LINKED ISSUE #131: Adjust NPC Dialog Box and add temporay dialog text
- Next button clickable,
- enter in keyboard
- etc.

### PR DESCRIPTION
**Pull request by VadimBhirawa**
### Main changes
Adjusted NPC Dialogue Box to be able to press enter to see the next line.

### Tasks List
- [close #131 ]

### Attachments
Adjust Dialogue Box

### CODE REVIEW

File: Assets/Astralization/Scripts/UI/DialogueInvoker.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+using System;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class DialogueInvoker : MonoBehaviour
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/UI/DialogueInvoker.cs:5]: add class docs

File: Assets/Astralization/Scripts/UI/DialogueManager.cs [hunk 2]
```diff
@@ -1,53 +1,125 @@
...
+
+    public string[] lines;
+    private int _index;
+    public float TextSpeed;
+
+    private PlayerInput _playerInput;
+    private InputAction _next;
+    private bool _dialogBoxOpen;
+
+    public static event Action<bool> DialogueEvent;
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/UI/DialogueManager.cs:31]: event name is too general, rename this to FinishDialogue

File: Assets/Astralization/Scripts/UI/DialogueInvoker.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+{
+    #region Variable
+    public static event Action DialogueInvoke;
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/UI/DialogueInvoker.cs:8]: event name is too general, rename this to StartDialogue

File: Assets/Astralization/Scripts/UI/DialogueManager.cs [hunk 4]
```diff
@@ -1,53 +1,125 @@
...
+using System.Collections;
+
+public interface IDialogueManager
+{
+    Animator GetAnimator();
+    bool GetDialogBox();
+    void ShowDialogBox(bool isInteractWithNpc);
+}
+
+public class DialogueManager : MonoBehaviour, IDialogueManager
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/UI/DialogueManager.cs:14]: add class docs

File: Assets/Astralization/Scripts/UI/DialogueManager.cs [hunk 5]
```diff
@@ -1,53 +1,125 @@
...
+        }
+        else
+        {
+            _animator.SetBool("IsOpen", false);
+            _dialogBoxOpen = false;
+            //Debug.Log("isInteractWithNpc: " + isInteractWithNpc);
+        }
+    }
+
+    void StartDialogue()
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/UI/DialogueManager.cs:94]: this function is not called when the dialogue is starting, rename this to SetUpDialogue

File: Assets/Astralization/Tests/NpcTest/NpcInteractableTest.cs [hunk 6]
```diff
@@ -41,10 +41,43 @@ public IEnumerator PlayerInteractableDetector_InteractNPC()
...
         Assert.IsTrue(npcController.GetState());
         Assert.AreEqual(playerInput.currentActionMap.ToString().Split(':')[1], "Dialogue");
         Assert.IsTrue(dialogueManager.GetAnimator().GetBool("IsOpen"));
     }
+
+    [UnityTest]
+    public IEnumerator PlayerInteractableDetector_NextLine()
+    {
+        yield return new WaitWhile(() => sceneLoaded == false);
+        GameObject player = GameObject.Find("Party/Iris");
```
Comment (rafiftaris) [inline Assets/Astralization/Tests/NpcTest/NpcInteractableTest.cs:55]: remove unused variables

File: Assets/Astralization/Tests/NpcTest/NpcInteractableTest.cs [hunk 7]
```diff
@@ -41,10 +41,43 @@ public IEnumerator PlayerInteractableDetector_InteractNPC()
...
+
+        INpcController npcController = npc.GetComponent<INpcController>();
+        PlayerInput playerInput = player.GetComponent<PlayerInput>();
+        IDialogueManager dialogueManager = dialogue.GetComponent<IDialogueManager>();
+
+        yield return SimulateInput(KeyboardMouseTestFixture.RegisteredInput.Interact);
+
+        yield return new WaitForSeconds(0.3f);
+        yield return SimulateInput(KeyboardMouseTestFixture.RegisteredInput.NextDialogueEnter);
+
```
Comment (rafiftaris) [inline Assets/Astralization/Tests/NpcTest/NpcInteractableTest.cs:74]: assert if the dialogue is updated

File: Assets/Astralization/Tests/NpcTest/NpcInteractableTest.cs [hunk 8]
```diff
@@ -41,10 +41,43 @@ public IEnumerator PlayerInteractableDetector_InteractNPC()
...
+        yield return new WaitForSeconds(0.3f);
+        yield return SimulateInput(KeyboardMouseTestFixture.RegisteredInput.NextDialogueEnter);
+
+        /*yield return new WaitForSeconds(0.3f);
```
Comment (rafiftaris) [inline Assets/Astralization/Tests/NpcTest/NpcInteractableTest.cs:78]: delete this

File: Assets/Astralization/Tests/NpcTest/NpcInteractableTest.cs [hunk 9]
```diff
@@ -41,10 +41,43 @@ public IEnumerator PlayerInteractableDetector_InteractNPC()
...
+        PlayerInput playerInput = player.GetComponent<PlayerInput>();
+        IDialogueManager dialogueManager = dialogue.GetComponent<IDialogueManager>();
+
+        yield return SimulateInput(KeyboardMouseTestFixture.RegisteredInput.Interact);
+
+        yield return new WaitForSeconds(0.3f);
+        yield return SimulateInput(KeyboardMouseTestFixture.RegisteredInput.NextDialogueEnter);
+
+        yield return new WaitForSeconds(0.3f);
+        yield return SimulateInput(KeyboardMouseTestFixture.RegisteredInput.NextDialogueEnter);
```
Comment (rafiftaris) [inline Assets/Astralization/Tests/NpcTest/NpcInteractableTest.cs:76]: change this to left click

**Review by rafiftaris — CHANGES_REQUESTED** [review]
- [BUG] when interact with npc for the 2nd time, dialogue is not resetted
- Add dialogue to feature docs, put it in HUD subsection
- Add NPC to feature docs, put it in Features section

**Review by rafiftaris — APPROVED** [review]
cool! 🔥

File: Assets/Astralization/Scripts/Interactables/Controller/NpcController.cs [hunk 10]
```diff
@@ -1,41 +1,41 @@
...
+    [Tooltip("True if NPC is interacted")]
+    private bool _isInteracted = false;
+
+    public static event Action<bool> NpcInteractionEvent;
+    #endregion
+
+    public override void OnInteraction()
+    {
+        _isInteracted = !_isInteracted;
+        //Debug.Log("OnInteraction: " + _isInteracted);
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/Interactables/Controller/NpcController.cs:27]: delete this debug, double with the one below

File: Assets/Astralization/Scripts/UI/DialogueManager.cs [hunk 11]
```diff
@@ -1,53 +1,137 @@
...
+        {
+            SetUpDialogue();
+            _animator.SetBool("IsOpen", true);
+            _dialogBoxOpen = true;
+        }
+        else
+        {
+            _animator.SetBool("IsOpen", false);
+            _dialogBoxOpen = false;
+            //Debug.Log("isInteractWithNpc: " + isInteractWithNpc);
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/UI/DialogueManager.cs:102]: remove this

File: Assets/Astralization/Scripts/UI/DialogueManager.cs [hunk 12]
```diff
@@ -1,53 +1,137 @@
...
+    [SerializeField]
+    private Text _nameText;
+    [SerializeField]
+    private Text _dialogueText;
+    [SerializeField]
+    private Animator _animator;
+
+    public string[] lines;
+    private int _index;
+    public float TextSpeed;
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/UI/DialogueManager.cs:31]: make private, add [SerializeField]

File: Assets/Astralization/Scripts/UI/DialogueManager.cs [hunk 13]
```diff
@@ -1,53 +1,137 @@
...
+
+    private PlayerInput _playerInput;
+    private InputAction _next;
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/UI/DialogueManager.cs:34]: unused both variables, remove

**Review by raihansyahdean — CHANGES_REQUESTED** [review]
Integrate ShowDialogueBox with Next function

**Review by raihansyahdean — APPROVED** [review]
Excellent!

**Pull request merged**